src/commands/kubeconfig.ts

```diff
@@ -0,0 +1,248 @@
+  // We'll set the context manually anyway, just in case. `aws update-kubeconfig` names the context
+  // with the EKS cluster's ARN.
+  try {
+    const kubectlResult = await exec(
+      "kubectl",
+      ["config", "use-context", awsClusterArn],
+      { check: true }
+    );
+    print2(kubectlResult.stdout);
+  } catch (error: any) {
+    print2("Failed to invoke `kubectl config use-context`");
+    throw error;
+  }
+
+  print2(
+    "Access granted and kubectl configured successfully. Re-run this command to refresh access if credentials expire."
+  );
+};
+
+/**
+ * Normalize the role argument to the format expected by the P0 backend,
+ * matching the way the Slack modal formats the role. Also validates that the
+ * role argument contains the components expected by the backend without having
+ * to make the API call.
+ *
+ * @param role The role argument to normalize
+ * @returns The normalized role value to pass to the backend
+ */
+const normalizeRoleArg = (role: string): string => {
+  const SEPARATOR = "/";
+  const SYNTAX_HINT =
+    "The role argument must be in one of the following formats:\n" +
+    "- ClusterRole/<roleName>\n" +
+    "- CuratedRole/<roleName>\n" +
+    "- Role/<namespace>/<roleName>";
+
+  const items = role.split(SEPARATOR).map((item) => item.trim());
+
+  if (items.length < 2 || items.length > 3) {
+    throw `Invalid format for role argument.\n${SYNTAX_HINT}`;
+  }
+
+  if (!items[0]) {
+    throw `Role kind must be specified.\n${SYNTAX_HINT}`;
+  }
+
+  if (stricmp(items[0], "ClusterRole")) {
+    return `ClusterRole ${SEPARATOR} ${items[1]}`;
+  } else if (stricmp(items[0], "CuratedRole")) {
+    return `CuratedRole ${SEPARATOR} ${items[1]}`;
+  } else if (stricmp(items[0], "Role")) {
+    if (items.length !== 3) {
+      throw `Invalid format for role argument.\n${SYNTAX_HINT}`;
+    }
+    return `Role ${SEPARATOR} ${items[1]} ${SEPARATOR} ${items[2]}`;
+  }
+
+  throw `Invalid role kind ${items[0]}.\n${SYNTAX_HINT}`;
+};
+
+/**
+ * Validate that the resource argument is of the format expected by the P0
+ * backend, again matching the way the Slack modal formats the resource.
+ *
+ * @param resource The resource argument to validate
+ */
+const validateResourceArg = (resource: string): void => {
+  const SEPARATOR = " / ";
+
+  const items = resource.split(SEPARATOR);
+
+  if (items.length < 2 || items.length > 3) {
+    throw (
+      "Invalid format for resource argument.\n" +
+      "The resource argument must be in one of the following formats (spaces required):\n" +
+      "- <kind> / <namespace> / <name>\n" +
+      "- <kind> / <name>"
+    );
+  }
+};
+
+const extractClusterNameAndRegion = (clusterArn: string) => {
+  const INVALID_ARN_MSG = `Invalid EKS cluster ARN: ${clusterArn}`;
+  // Example EKS cluster ARN: arn:aws:eks:us-west-2:123456789012:cluster/my-testing-cluster
+  const parts = clusterArn.split(":");
```

Comment:
ARNs come in different formats, should we use a function like https://github.com/p0-security/app/blob/main/backend/src/integrations/resources/aws/utils.ts#L83

or are we guaranteed to always get this format?